includes/Products/Sync/Background.php

```diff
@@ -139,9 +145,115 @@ public function process_items( $job, $data, $items_per_batch = null ) {
 	 * @param mixed $item
 	 * @param object|\stdClass $job
 	 * @return array
+	 * @throws Framework\SV_WC_Plugin_Exception
 	 */
-	public function process_item($item, $job) {
-		// TODO
+	public function process_item( $item, $job ) {
+
+		list( $product_id, $method ) = $item;
+
+		$product = wc_get_product( $product_id );
+
+		if ( ! $product instanceof \WC_Product ) {
+			throw new Framework\SV_WC_Plugin_Exception( "No product found with ID equal to {$product_id}." );
+		}
+
+		if ( ! in_array( $method, [ Sync::ACTION_UPDATE, Sync::ACTION_DELETE ], true ) ) {
+			throw new Framework\SV_WC_Plugin_Exception( "Invalid sync request method: {$method}." );
+		}
+
+		if ( Sync::ACTION_UPDATE === $method ) {
+			$request = $this->process_item_update( $product );
+		} else {
+			$request = $this->process_item_delete( $product );
+		}
+
+		return $request;
+	}
+
+
+	/**
+	 * Processes an UPDATE sync request for the given product.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param \WC_Product $product product object
+	 */
+	private function process_item_update( $product ) {
+
+		$product_data = $this->prepare_product_data( $product );
+
+		return [
+			'retailer_id' => $product_data['retailer_id'],
+			'method'      => Sync::ACTION_UPDATE,
+			'data'        => $product_data,
+		];
+	}
+
+
+	/**
+	 * Prepares the product data to be included in a sync request.
+	 *
+	 * @since 2.0.0-dev.1
+	 *
+	 * @param \WC_Product $product product object
+	 * @return array
+	 * @throws Framework\SV_WC_Plugin_Exception
+	 */
+	private function prepare_product_data( $product ) {
+
+		if ( $product->is_type( 'variation' ) ) {
+
+			$parent_product = wc_get_product( $product->get_parent_id() );
+
+			if ( ! $parent_product instanceof \WC_Product ) {
+				throw new Framework\SV_WC_Plugin_Exception( "No parent product found with ID equal to {$product->get_parent_id()}." );
+			}
+
+			$fb_parent_product = new \WC_Facebook_Product( $parent_product );
+			$fb_product        = new \WC_Facebook_Product( $product->get_id(), $fb_parent_product );
+
+			$retailer_product_group_id = \WC_Facebookcommerce_Utils::get_fb_retailer_id( $parent_product );
+
+		} else {
+
+			$fb_product = new \WC_Facebook_Product( $product->get_id() );
+
+			$retailer_product_group_id = null;
+		}
+
+		$data = $fb_product->prepare_product();
+
+		// allowed values are 'refurbished', 'used', and 'new', but the plugin has always used the latter
+		$data['condition'] = 'new';
+
+		$data['product_type'] = $data['category'];
+
+		$data['retailer_product_group_id'] = $retailer_product_group_id ?: $data['retailer_id'];
+
+		// attributes other than size, color, pattern, or gender need to be included in the additional_variant_attributes field
+		if ( isset( $data['custom_data'] ) && is_array( $data['custom_data'] ) ) {
+
+			$data['additional_variant_attributes'] = $data['custom_data'];
+			unset( $data['custom_data'] );
+		}
+
+		return $data;
```

Comment:
May be a good spot to add a filter. Especially with the harcoded `condition`, even if it has always been done that way it wouldn't hurt to give shops an option to change it and other data if needed

Reply:
Actually on second thought, it may make more sense to put the filter in `process_item_update` and another matching filter in `process_item_delete` instead.

Reply:
That's good idea. I added the filters in d30612d, 8a4ae86, and tests in e3fabe7.